src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/domainevents/DomainEventDescriberTest.kt

```diff
@@ -283,6 +285,26 @@ class DomainEventDescriberTest {
     assertThat(result).isEqualTo("A request for placement was withdrawn for dates Tuesday 02 January 2024 to Monday 04 March 2024, Monday 06 May 2024 to Monday 08 July 2024")
   }
 
+  @Test
+  fun `Returns expected description for match request withdrawn event`() {
+    val domainEventSummary = DomainEventSummaryImpl.ofType(DomainEventType.APPROVED_PREMISES_MATCH_REQUEST_WITHDRAWN)
+
+    every { mockDomainEventService.getMatchRequestWithdrawnEvent(UUID.fromString(domainEventSummary.id)) } returns buildDomainEvent {
+      MatchRequestWithdrawnEnvelope(
+        id = it,
+        timestamp = Instant.now(),
+        eventType = "approved-premises.match-request.withdrawn",
+        eventDetails = MatchRequestWithdrawnFactory()
+          .withDatePeriod(DatePeriod(LocalDate.of(2024, 1, 2), LocalDate.of(2024, 3, 4)))
+          .produce(),
+      )
+    }
+
+    val result = domainEventDescriber.getDescription(domainEventSummary)
+
+    assertThat(result).isEqualTo("A request for placement was withdrawn for dates Tuesday 02 January 2024 to Monday 04 March 2024")
```

Comment:
It'd be good if we could update `cas1UiExtendedDateFormat` to remove leading zeros

Reply:
ohhh, good call! done